tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_ms.py

```diff
@@ -0,0 +1,285 @@
+import logging
+import pytest
+
+from ocs_ci.framework.testlib import (
+    tier4,
+    tier4a,
+    ManageTest,
+    managed_service_required,
+    ignore_leftovers,
+)
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import machine, constants, cephfs_workload
+from ocs_ci.ocs.resources.pod import (
+    wait_for_pods_to_be_in_statuses,
+    check_pods_after_node_replacement,
+)
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+from ocs_ci.ocs.node import (
+    get_osd_running_nodes,
+    get_worker_nodes,
+    get_node_objs,
+    get_nodes,
+    get_node_osd_ids,
+    wait_for_nodes_status,
+    recover_node_to_ready_state,
+    label_nodes,
+    get_ocs_nodes,
+    get_osd_ids_per_node,
+    get_node_rook_ceph_pod_names,
+    verify_worker_nodes_security_groups,
+    wait_for_osd_ids_come_up_on_node,
+    wait_for_all_osd_ids_come_up_on_nodes,
+)
+
+log = logging.getLogger(__name__)
+
+
+def get_node_pod_names_expect_to_terminate(node_name):
+    """
+    Get the node pod names expected to be in a Terminating state
+
+    Args:
+        node_name (str): The node name
+
+    Returns:
+        list: list of the node pod names expected to be in a Terminating state
+
+    """
+    node_rook_ceph_pod_names = get_node_rook_ceph_pod_names(node_name)
+    return [
+        pod_name
+        for pod_name in node_rook_ceph_pod_names
+        if not pod_name.startswith(("rook-ceph-operator", "rook-ceph-osd-prepare"))
+    ]
+
+
+def get_all_pod_names_expect_to_terminate():
```

Comment:
```suggestion
def get_all_pod_names_expected_to_terminate():
```
Just a suggestion - not mandatory.